src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -1,28 +1,35 @@
 use crate::{
     buffers::Acker,
-    event::{Event, Metric},
+    event::{metric::Direction, Event, Metric},
     region::RegionOrEndpoint,
     sinks::util::{
         retries::{FixedRetryPolicy, RetryLogic},
-        BatchServiceSink, Partition, SinkExt,
+        BatchServiceSink, SinkExt,
     },
     topology::config::{DataType, SinkConfig},
 };
+use chrono::{DateTime, SecondsFormat, Utc};
 use futures::{Future, Poll};
 use rusoto_cloudwatch::{
     CloudWatch, CloudWatchClient, MetricDatum, PutMetricDataError, PutMetricDataInput,
 };
 use rusoto_core::{Region, RusotoFuture};
 use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
 use std::{convert::TryInto, time::Duration};
 use tower::{Service, ServiceBuilder};
-use tracing::field;
 use tracing_futures::{Instrument, Instrumented};
 
+#[derive(Clone, Default)]
+struct CloudWatchMetricsState {
+    gauges: HashMap<String, f64>,
+}
+
 #[derive(Clone)]
 pub struct CloudWatchMetricsService {
```

Comment:
take it or leave it: but in cloudwatch logs we usually name the struct that implements service CloudwatchLogsSvc instead of Service to indicate that it is just an implementation not a trait. I'm fine with keeping it how it is now, just wanted to make sure you were aware.

Reply:
Thanks!